src/WireBST.java

```diff
@@ -109,29 +109,242 @@ private void insertToRight(WireBSTNode currentNode , WireBSTNode newNode ){
         newNode.setLeft(currentNode); // make the predecessor of the new node to be her parent, because it is the predecessor (only in insert to left!).
     }
 
+    /**
+     * update the median after insert a node
+     * @param node, the node that inserted
+     */
     private void updateMedianInsert(WireBSTNode newNode){
-        if(newNode.getStudentNum() > this.getMedianWireBST().getStudentNum() ){
-            if (this.getNodeCounter() % 2 == 0){
-                this.setMedian ( getSuccessor( this.getMedianWireBST() ) ) ;
-            }
-        else
-            if (this.getNodeCounter() % 2 == 1){
-                this.setMedian ( getPredecessor( this.getMedianWireBST() ) ) ;
+        if(newNode.getStudentNum() > this.getMedianWireBST().getStudentNum() ) {
+            if (this.getNodeCounter() % 2 == 0) {
+                this.setMedian(getSuccessor(this.getMedianWireBST()));
             }
+        }
+        else if (this.getNodeCounter() % 2 == 1) {
+            this.setMedian(getPredecessor(this.getMedianWireBST()));
+        }
         this._nodeCounter ++;
+    }
+
+    /**
+     * remove the student that is number is the given parameter
+     * @param studentNum, the node who include that number
+     * @return true if the node was removed, else false
+     */
+    public boolean removeWireBSTNode(int studentNum){
+
+        WireBSTNode temp = searchWireBST( studentNum );
+
+        if (temp == null){
+
+            return false;
+        }
+
+        else {
+
```

Comment:
need to add:
updateMedianRemove(node);

Reply:
and remove all the other duplicated sub-calls to this in the sub-functions of remove